pkg/cli/clients/management.go

```diff
@@ -363,111 +355,94 @@ func (amc *UCPApplicationsManagementClient) CreateApplicationIfNotFound(ctx cont
-	obj, found := resource.Properties["environment"]
-	// A resource may not have an environment associated with it.
-	if !found {
-		return false
+	client, err := amc.createApplicationClient(scope)
+	if err != nil {
+		return false, err
 	}
 
-	associatedEnvId, ok := obj.(string)
-	if !ok || associatedEnvId == "" {
-		return false
-	}
+	var response *http.Response
+	ctx = amc.captureResponse(ctx, &response)
 
-	idParsed, err := resources.ParseResource(associatedEnvId)
+	_, err = client.Delete(ctx, name, nil)
 	if err != nil {
-		return false
-	}
-
-	if strings.EqualFold(idParsed.Name(), environmentName) {
-		return true
+		return false, err
 	}
 
-	return false
+	return response.StatusCode != 204, nil
 }
 
-// ListEnvironmentsInResourceGroup creates a list of environment resources by paging through the list of environments in
-// the resource group and appending each environment to the list. It returns the list of environment resources or an error
-// if one occurs.
-func (amc *UCPApplicationsManagementClient) ListEnvironmentsInResourceGroup(ctx context.Context) ([]corerpv20231001.EnvironmentResource, error) {
-	envResourceList := []corerpv20231001.EnvironmentResource{}
-
-	envClient, err := corerpv20231001.NewEnvironmentsClient(amc.RootScope, &aztoken.AnonymousCredential{}, amc.ClientOptions)
+// ListEnvironments lists all environments in the configured scope (assumes configured scope is a resource group).
+func (amc *UCPApplicationsManagementClient) ListEnvironments(ctx context.Context) ([]corerpv20231001.EnvironmentResource, error) {
+	client, err := amc.createEnvironmentClient(amc.RootScope)
 	if err != nil {
-		return envResourceList, err
+		return []corerpv20231001.EnvironmentResource{}, err
 	}
 
-	pager := envClient.NewListByScopePager(&corerpv20231001.EnvironmentsClientListByScopeOptions{})
+	environments := []corerpv20231001.EnvironmentResource{}
+	pager := client.NewListByScopePager(&corerpv20231001.EnvironmentsClientListByScopeOptions{})
 	for pager.More() {
-		nextPage, err := pager.NextPage(ctx)
+		page, err := pager.NextPage(ctx)
 		if err != nil {
-			return envResourceList, err
+			return []corerpv20231001.EnvironmentResource{}, err
 		}
-		applicationList := nextPage.EnvironmentResourceListResult.Value
-		for _, application := range applicationList {
-			envResourceList = append(envResourceList, *application)
+
+		for _, environment := range page.EnvironmentResourceListResult.Value {
+			environments = append(environments, *environment)
 		}
 	}
 
-	return envResourceList, nil
+	return environments, nil
 }
 
 // ListEnvironmentsAll queries the scope for all environment resources and returns a slice of environment resources or an error if one occurs.
 func (amc *UCPApplicationsManagementClient) ListEnvironmentsAll(ctx context.Context) ([]corerpv20231001.EnvironmentResource, error) {
-	scope, err := resources.ParseScope("/" + amc.RootScope)
+	scope, err := resources.ParseScope(amc.RootScope)
```

Comment:
Why did we remove `"/"`? Was this unnecessary?

Reply:
The generated clients don't like it when the scope has a leading slash. We used to initialize the client without the leading slash (`planes/radius/local/...` instead of `/planes/radius/local/...`)

I moved the code that trims the leading slash to the place where the generated clients are constructed for clarity. It's weird to have the "root scope" in a field without it being in the correct format.